tests/vmi_headless_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017, 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	kubev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("[rfe_id:609]VMIheadless", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var vmi *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		vmi = tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskAlpine))
+	})
+
+	Describe("[rfe_id:273]Creating a VirtualMachineInstance", func() {
+
+		Context("with headless", func() {
+
+			BeforeEach(func() {
+				f := false
+				vmi.Spec.Domain.Devices.AutoattachGraphicsDevice = &f
+			})
+
+			It("should create headless vmi without any issue", func() {
+				tests.RunVMIAndExpectLaunch(vmi, 30)
+			})
+
+			It("[test_id:714][posneg:positive]should not have vnc graphic device in xml", func() {
+				tests.RunVMIAndExpectLaunch(vmi, 30)
+
+				runningVMISpec, err := tests.GetRunningVMISpec(vmi)
+				Expect(err).ToNot(HaveOccurred(), "should get vmi spec without problem")
+
+				Expect(len(runningVMISpec.Devices.Graphics)).To(Equal(0), "should not have any graphics devices present")
+			})
+
+			It("[test_id:737][posneg:positive]should match memory with overcommit enabled", func() {
```

Comment:
We also test in https://github.com/kubevirt/kubevirt-ansible/blob/master/tests/high_performance_vm_test.go
that can be moved to here, since it handles headless and over commit.

Reply:
I I have updated the tests. The second 737 is the same test, but negative. I removed the ID and left it. IMO it is important to test both sides positive and negative.

Reply:
so we will need to add now case in Polarion for the Negative case, will do it.